workflow/type.go

```diff
@@ -22,23 +22,32 @@ const (
 // Workflow describes a workflow of a service
 type Workflow struct {
 	Hash    hash.Hash `hash:"-" validate:"required"`
-	Trigger *Trigger  `hash:"name:1" validate:"required"`
-	Tasks   []*Task   `hash:"name:2" validate:"required"`
-	Key     string    `hash:"name:3" validate:"required"`
+	Key     string    `hash:"name:1" validate:"required"`
+	Trigger Trigger   `hash:"name:2" validate:"required"`
+	Nodes   []Node    `hash:"name:3" validate:"dive,required"`
+	Edges   []Edge    `hash:"name:4" validate:"dive,required"`
 }
 
-// Task describes the instructions for the workflow to execute a task
-type Task struct {
-	InstanceHash hash.Hash `hash:"name:1" validate:"required"`
-	TaskKey      string    `hash:"name:2" validate:"printascii"`
+// Node describes the instructions for the workflow to execute a task
+type Node struct {
+	Key          string    `hash:"name:1" validate:"required"`
+	InstanceHash hash.Hash `hash:"name:2" validate:"required"`
+	TaskKey      string    `hash:"name:3" validate:"required,printascii"`
+}
+
+// Edge describes the instructions for the workflow to execute a task
+type Edge struct {
+	Src string `hash:"name:1" validate:"required"`
```

Comment:
I don't see a validation if the edge has proper src and dst.

Reply:
yes not yet, I wanted to add the validation of nodes/edges and also the instances on another pr